src/Microsoft.Health.Fhir.Shared.Api/Features/Filters/OperationOutcomeExceptionFilterAttribute.cs

```diff
@@ -79,6 +80,19 @@ public override void OnActionExecuted(ActionExecutedContext context)
                         operationOutcomeResult.StatusCode = HttpStatusCode.ServiceUnavailable;
                         break;
                     case ResourceNotValidException _:
+                        if (context.ActionDescriptor is ControllerActionDescriptor)
+                        {
+                            var controllerDescriptor = (ControllerActionDescriptor)context.ActionDescriptor;
+                            if (controllerDescriptor.ControllerTypeInfo.Name == "ValidateController")
```

Comment:
I am not familiar with the `ControllerActionDescriptor` class, but can we use `ControllerName` here instead of `ControllerTypeInfo.Name`

Reply:
Good catch, changed.